src/zcash/ZcashEngine.ts

```diff
@@ -442,13 +442,26 @@ export class ZcashEngine extends CurrencyEngine<
     if (eq(nativeAmount, '0')) throw new NoAmountSpecifiedError()
 
     const synchronizer = await this.synchronizerPromise
-    const proposal = await synchronizer.proposeTransfer({
-      toAddress: publicAddress,
-      zatoshi: nativeAmount,
-      memo: memos[0]?.value ?? ''
-    })
+    // If a ZIP-321 Payment URI is provided, use that flow instead:
+    const zip321Uri: string | undefined = (edgeSpendInfo.otherParams as any)
+      ?.zip321Uri
+    if (zip321Uri != null) {
+      this.log(`[ZEC propose] zip321Uri ${zip321Uri}`)
+    }
+
+    const proposal =
+      zip321Uri != null && synchronizer?.proposeFulfillingPaymentURI != null
```

Comment:
The `synchronizer?.proposeFulfillingPaymentURI != null` is unnecessary as it's required in the type